lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -211,16 +208,19 @@ class QcFlagService {
             });
             if (!targetRun) {
             // eslint-disable-next-line max-len
-                throw new BadParameterError(`You cannot insert flag for simulation pass (id:${simulationPassId}), run (runNumber:${runNumber}), detector (name:${dplDetector.name}) as there is no association between them`);
+                throw new BadParameterError(`There is not association between simulation pass with this id (${simulationPassId}), run with this number (${runNumber}) and detector with this name (${dplDetector.name})`);
+            }
+
+            const fromTime = from ?? targetRun.startTime;
+            const toTime = to ?? targetRun.endTime;
+
+            if (fromTime >= toTime) {
+                throw new BadParameterError('Parameter "to" timestamp must be greater than "from" timestamp');
             }
 
-            const isFromTimeBeyondReange = fromTime
-                && (fromTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < fromTime);
-            const isToTimeBeyondReange = toTime
-                && (toTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < toTime);
-            if (isFromTimeBeyondReange || isToTimeBeyondReange) {
+            if (fromTime < targetRun.startTime || targetRun.endTime < toTime) {
                 // eslint-disable-next-line max-len
-                throw new BadParameterError(`Given QC flag period (${fromTime} ${toTime}) is beyond run trigger period (${targetRun.timeTrgStart.getTime()}, ${targetRun.timeTrgEnd.getTime()})`);
+                throw new BadParameterError(`Given QC flag period (${fromTime}, ${toTime}) is out of run trigger period (${targetRun.startTime}, ${targetRun.endTime})`);
             }
```

Comment:
Maybe you can factorize this, at least validation for example?

Reply:
factorized